oauth-common/src/main/java/io/strimzi/kafka/oauth/metrics/MetricKey.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2017-2021, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.kafka.oauth.metrics;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+public class MetricKey {
```

Comment:
It would actually be better named SensorKey, but that will require renaming a whole bunch of classes to keep it consistent.

Do we have some policy on when to add JavaDoc and when not? Not all public classes should be considered a public API do we have a policy to mark documented classes as not considered a public API?

Reply:
> that will require renaming a whole bunch of classes to keep it consistent.

But your IDE makes refactoring really easy, so if there are better names I don't understand why such renaming is a barrier.

> Do we have some policy on when to add JavaDoc and when not?

I don't see the purpose of Javadoc being always documentation for public consumption. It's a really good way of recording what the purpose of the class is, what invariants/assumptions/guarantees it has/makes etc. The ultimate user of this documentation will be someone maintaining this code in the future (most likely you), and anyone else who wants to understand how the classes fit together (me, or other reviewers, for example).

Reply:
> But your IDE makes refactoring really easy, so if there are better names I don't understand why such renaming is a barrier.

Never mind, it wasn't meant that way.

> I don't see the purpose of Javadoc being always documentation for public consumption. It's a really good way of recording what the purpose of the class is, what invariants/assumptions/guarantees it has/makes etc. The ultimate user of this documentation will be someone maintaining this code in the future (most likely you), and anyone else who wants to understand how the classes fit together (me, or other reviewers, for example).

I'll add more JavaDoc, and very nice you pointed out what exactly you would like to be explained.

Reply:
Thanks!

Reply:
Added javadoc. Renamed class to SensorKey.